index.js

```diff
@@ -0,0 +1,91 @@
+let slideIndex = 0;
+
+function moveSlide(n) {
+  const slides = document.querySelectorAll('.carousel__item');
+  const totalSlides = slides.length;
+
+  slideIndex += n;
+
+  if (slideIndex >= totalSlides) {
+    slideIndex = 0;
+  } else if (slideIndex < 0) {
+    slideIndex = totalSlides - 1;
+  }
+
+  updateSlidePosition();
+  updateDots();
+}
+
+function currentSlide(n) {
+  slideIndex = n;
+  updateSlidePosition();
+  updateDots();
+}
+
+function updateSlidePosition() {
+  const slides = document.querySelectorAll('.carousel__item');
+  const slideWidth = slides[0].offsetWidth;
+  const offset = -slideIndex * slideWidth;
+  document.querySelector('.carousel__img').style.transform = `translateX(${offset}px)`;
+}
+
+function updateDots() {
+  const dots = document.querySelectorAll('.dot');
```

Comment:
Similar al comentario anterior, usa clases `js-`
```suggestion
  const dots = document.querySelectorAll('.js-dot');
```